src/ciphersuite/ciphersuites.rs

```diff
@@ -14,24 +14,27 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see http://www.gnu.org/licenses/.
 
-use crate::ciphersuite::*;
+use crate::{ciphersuite::*, errors::Error};
+pub(crate) use std::convert::TryFrom;
 
 impl From<&CiphersuiteName> for u16 {
     fn from(s: &CiphersuiteName) -> u16 {
         *s as u16
     }
 }
 
-impl From<u16> for CiphersuiteName {
-    fn from(v: u16) -> Self {
+impl TryFrom<u16> for CiphersuiteName {
+    type Error = Error;
+
+    fn try_from(v: u16) -> Result<Self, Self::Error> {
         match v {
-            0x0001 => CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519,
-            0x0002 => CiphersuiteName::MLS10_128_DHKEMP256_AES128GCM_SHA256_P256,
-            0x0003 => CiphersuiteName::MLS10_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519,
-            0x0004 => CiphersuiteName::MLS10_256_DHKEMX448_AES256GCM_SHA512_Ed448,
-            0x0005 => CiphersuiteName::MLS10_256_DHKEMP521_AES256GCM_SHA512_P521,
-            0x0006 => CiphersuiteName::MLS10_256_DHKEMX448_CHACHA20POLY1305_SHA512_Ed448,
-            _ => panic!("Not implemented."),
+            0x0001 => Ok(CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519),
+            0x0002 => Ok(CiphersuiteName::MLS10_128_DHKEMP256_AES128GCM_SHA256_P256),
+            0x0003 => Ok(CiphersuiteName::MLS10_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519),
+            0x0004 => Ok(CiphersuiteName::MLS10_256_DHKEMX448_AES256GCM_SHA512_Ed448),
+            0x0005 => Ok(CiphersuiteName::MLS10_256_DHKEMP521_AES256GCM_SHA512_P521),
+            0x0006 => Ok(CiphersuiteName::MLS10_256_DHKEMX448_CHACHA20POLY1305_SHA512_Ed448),
+            _ => Err(Error::DecodingError),
```

Comment:
Do we explicitly want errors when fuzzing as opposed to panics? Or What is the motivation for this change?

Reply:
We generally don't want to panic. Fuzzing is running into all the error cases. Panicking will be an error.